src/MappedArrays.jl

```diff
@@ -62,9 +62,10 @@ parenttype{T,N,A,F,Finv}(::Type{MappedArray{T,N,A,F,Finv}}) = A
 
 function testvalue(data)
     if !isempty(data)
-        return first(data)
-    end
-    zero(eltype(data))
+        first(data)
+    else
+        zero(eltype(data))
+    end::eltype(data)
```

Comment:
I see. But if I understand this correctly this code isn't about inference but about making the concrete array it returns to the tighest `eltype` fit it can, yes? If so we are set already I think.

```julia
julia> data = Real[3.2]
1-element Array{Real,1}:
 3.2

julia> mappedarray(round, data)
1-element MappedArrays.ReadonlyMappedArray{Float64,1,Array{Real,1},Base.#round}:
 3.0

julia> eltype(mappedarray(round, data))
Float64

julia> supertype(typeof(mappedarray(round,data)))
MappedArrays.AbstractMappedArray{Float64,1}
```